src/ocean/util/container/TreeMap.d

```diff
@@ -0,0 +1,610 @@
+/*******************************************************************************
+
+    EBtree based map.
+
+    By default the map uses plain `eb64_node` nodes.
+    To attach values to an ebtree node, create a struct, say, `Node`, whose
+    first field is of type `eb64_node`, followed by any number of other fields,
+    and pass this struct as the `TreeMap` template parameter.
+      - Do not manipulate the first `eb64_node` field (or the plain node if not
+        passing an own struct). Reading may be useful as it contains the key.
+      - Instances of this struct handed out by `TreeMap` are malloc-allocated so
+        they are invisible to the GC.
+
+    We are using an ebtree with malloc-allocated nodes, rather than a hash map,
```

Comment:
This paragraph appears to be written from the perspective of something internal in swarm. It would make more sense here rephrased as "pros and cons vs HashMap".